pkg/acsengine/defaults.go

```diff
@@ -618,7 +618,7 @@ func setAgentProfileDefaults(a *api.Properties, isUpgrade, isScale bool) {
 		// don't default Distro for OpenShift
 		if !a.OrchestratorProfile.IsOpenShift() {
 			if profile.Distro == "" {
-				if a.HostedMasterProfile != nil {
+				if profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < api.VHDDiskSizeAKS {
```

Comment:
- `profile.OSDiskSizeGB == 0` we interpret as "no user configuration provided, i.e., zero value"; we can use the 100GB VHD image in this condition
- for explicit OS disk sizes < 100 GB we need to use the base Ubuntu OS image, as we can't use a VHD image reference whose OS disk size is greater than the configured disk size in the ARM template